examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/DiscoveryExampleFragment.java

```diff
@@ -0,0 +1,390 @@
+          arrayAdapter.clear();
+        });
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    Log.i(TAG, "onResume() called. Calling startDiscovery()");
+    if (!startDiscovery(15)) {
+      Log.e(TAG, "onResume() Warning: startDiscovery() call Failed");
+    }
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    Log.i(TAG, "onPause() called");
+    stopDiscovery();
+    scheduledFutureTask.cancel(true);
+  }
+
+  /** Interface for notifying the host. */
+  public interface Callback {
+    /** Notifies listener of Connection Button click. */
+    // TODO: In following PRs. Implement CastingPlayer connection
+    void handleConnectionButtonClicked(CastingPlayer castingPlayer);
+  }
+
+  private boolean startDiscovery(int discoveryDuration) {
+    Log.i(TAG, "startDiscovery() called, discoveryDuration: " + discoveryDuration + " seconds");
+    if (discovering) {
+      Log.e(TAG, "startDiscovery() Warning: already discovering, stop before starting");
+      return false;
+    }
+    // Add the implemented CastingPlayerChangeListener to listen to changes in the discovered
+    // CastingPlayers
+    MatterError errAdd =
+        matterCastingPlayerDiscovery.addCastingPlayerChangeListener(castingPlayerChangeListener);
+    if (errAdd.hasError()) {
+      Log.e(TAG, "startDiscovery() addCastingPlayerChangeListener() called, errAdd: " + errAdd);
+      return false;
+    }
+    // Start discovery
+    Log.i(TAG, "startDiscovery() calling startDiscovery()");
+    MatterError errStart = matterCastingPlayerDiscovery.startDiscovery();
+    if (errStart.hasError()) {
+      Log.e(TAG, "startDiscovery() startDiscovery() called, errStart: " + errStart);
+      return false;
+    }
+
+    discovering = true;
+    discoveryAttempt++;
+    Log.i(TAG, "startDiscovery() started discovery attempt #" + discoveryAttempt);
+
+    matterDiscoveryMessageTextView.setText(
+        getString(R.string.matter_discovery_message_discovering_text));
+    Log.d(
+        TAG,
+        "startDiscovery() text set to: "
+            + getString(R.string.matter_discovery_message_discovering_text));
+
+    // Schedule a service to stop discovery and remove the CastingPlayerChangeListener
+    // Safe to call if discovery is not running
+    scheduledFutureTask =
+        executorService.schedule(
+            () -> {
+              Log.i(
+                  TAG,
+                  "startDiscovery() executorService "
+                      + discoveryDuration
+                      + " seconds timer expired. Calling stopDiscovery() on attempt #"
+                      + discoveryAttempt);
+              stopDiscovery();
+            },
+            discoveryDuration,
+            TimeUnit.SECONDS);
+
+    return true;
+  }
+
+  private void stopDiscovery() {
+    Log.i(TAG, "stopDiscovery() called on attempt #" + discoveryAttempt);
+    boolean stopDiscoverySuccess = true;
```

Comment:
I don't think we need to track this stopDiscoverySuccess, since the states are already being tracked in the tv-casting-common layer.